File: LeetCode/1299.cpp
//Link: https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/

class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int greatest = -1, temp;
        for(int i = arr.size()-1 ; i >= 0 ; i--) {
            temp = arr[i];
            arr[i] = greatest;
            greatest = max(greatest, temp);
        }
        return arr;
    }
};

File: LeetCode/py/1299.py
#Link: https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/

class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        temp = greatest = -1
        for i in range(len(arr)-1 , -1 , -1):
            temp = arr[i]
            arr[i] = greatest
            greatest = max(greatest, temp)
        return arr
        